src/code/audio/audio_seqplayer.c

```diff
@@ -1,73 +1,2314 @@
+                    } else if (cmd == 0x7F) {
+                        // Drums
+                        layer->instOrWave = 0;
+                    } else {
+                        // Synthetic Wave
+                        layer->instOrWave = cmd;
+                        layer->instrument = NULL;
+                    }
+
+                    if (cmd == 0xFF) {
+                        layer->adsr.decayIndex = 0;
+                    }
+                } else {
+                    // Instrument
+                    if ((layer->instOrWave =
+                             AudioSeq_GetInstrumentId(channel, cmd, &layer->instrument, &layer->adsr)) == 0) {
+                        layer->instOrWave = 0xFF;
+                    }
+                }
+                break;
+
+            case 0xC7: // layer: enable portamento
+                layer->portamento.mode = AudioSeq_ScriptReadU8(state);
+
+                cmd = AudioSeq_ScriptReadU8(state);
+                cmd += channel->transposition;
+                cmd += layer->transposition;
+                cmd += seqPlayer->transposition;
+
+                if (cmd >= 0x80) {
+                    cmd = 0;
+                }
+
+                layer->portamentoTargetNote = cmd;
+
+                // If special, the next param is u8 instead of var
+                if (PORTAMENTO_IS_SPECIAL(layer->portamento)) {
+                    layer->portamentoTime = *(state->pc++);
+                    break;
+                }
+
+                cmdArg16 = AudioSeq_ScriptReadCompressedU16(state);
+                layer->portamentoTime = cmdArg16;
+                break;
+
+            case 0xC8: // layer: disable portamento
+                layer->portamento.mode = PORTAMENTO_MODE_OFF;
+                break;
+
+            case 0xCB: // layer: set envelope and decay index
+                cmdArg16 = AudioSeq_ScriptReadS16(state);
+                layer->adsr.envelope = (AdsrEnvelope*)(seqPlayer->seqData + cmdArg16);
+                // fallthrough
+            case 0xCF: // layer: set decay index
+                layer->adsr.decayIndex = AudioSeq_ScriptReadU8(state);
+                break;
+
+            case 0xCC: // layer: ignore drum pan
+                layer->ignoreDrumPan = true;
+                break;
+
+            case 0xCD: // layer: stereo effects
+                layer->stereo.asByte = AudioSeq_ScriptReadU8(state);
+                break;
+
+            case 0xCE: // layer: bend pitch
+                cmdArg8 = AudioSeq_ScriptReadU8(state);
+
+                layer->bend = gBendPitchTwoSemitonesFrequencies[(u8)(cmdArg8 + 0x80)];
+                break;
+
+            case 0xF0: // layer:
+                cmdArg16 = AudioSeq_ScriptReadS16(state);
+
+                layer->unk_0A.asByte &= (cmdArg16 ^ 0xFFFF);
+                break;
+
+            case 0xF1: // layer:
+                layer->unk_08 = AudioSeq_ScriptReadU8(state);
+                break;
```

Comment:
Are these unknown or unused?

Reply:
unknown. More audio docs is needed to know what these commands do. I haven't looked or extracted any of MM's sequences yet, so I have no clue what could be used or unused